Refactoring

Stan-Kudri opened the pull request.

Stan-Kudri committed:
Changed: Auto-Implemented Properties X and Y.

Stan-Kudri committed:
Changed:

-Main: Add a method "Game" in which the launch and implementation of the game is performed.

-Direct:
  1)Renamed the field from _directory to _currentDirection.
  2)Rename method from NewDirection to ChangeDirection and input parameter type.
-Field:
  1)Replaced size fields to AutoProperty.
-Snake:
  1)Changed field size assignment.
  2)Added a method for creating the body of a snake, which was taken out of the constructor.

Stan-Kudri committed:
Changed the conditions for the error and its message.

Stan-Kudri committed:
Add method WriteMessage (Game Over) and changed method Move() and TheEnd in class Snake.

blowin commented on Snake/Snake.cs:
```diff
@@ -52,6 +42,8 @@ public class Snake {
             _head.Draw();
 
             Thread.Sleep(100);
```
Это так и не убрал, а если я скажу протестировать змейку, каждый тест будет становиться медленнее в зависимости от количества вызовов этото метода

blowin commented on Snake/Snake.cs:
```diff
@@ -85,9 +77,25 @@ public class Snake {
         }
 
-        private bool TheEnd(Point point) {
+        private void BuildTheSnakeBody(int x, int y) {
```
BuildBody (ты пишешь метод внутри Snake, так что понятно чьё тело)

blowin commented on Snake/Snake.cs:
```diff
@@ -85,9 +77,25 @@ public class Snake {
+            _head = _body.Last();
+        }
+
+        private bool TheEnd(Point movePoint) {
```
Метод остался с тем же названием

blowin commented:
Почему задача рефакторинг, а ветка с названием изменения авто свойств?

Stan-Kudri committed:
Changed:
-Rendered "Thread.Sleep(100)" to a method "Game".
- Rename methods:
  1)BuildTheSnakeBody -> BuildBody.
  1)TheEnd-> ClashPoint.

Stan-Kudri committed:
Changed:
-Add one item enum Directions : Other;
-Changed the method "ChangeDirection" to an implementation with an input type parameter Directions.
-Main:
  1)Add method Directions GetDirection(ConsoleKey key);
  2)Changed the method "Game" using method "GetDirection".

Stan-Kudri committed:
Add empty lines after conditions (if).

Stan-Kudri committed:
Add folder Extension and moved the method "GetDirection".

Stan-Kudri committed:
Removed the field "_tail" and made the creation of a variable in the method "Move", where is it used.

Stan-Kudri committed:
Add empty line.

blowin commented on Snake/Snake.cs:
```diff
@@ -90,7 +88,7 @@ public class Snake {
         }
 
-        private bool TheEnd(Point movePoint) {
+        private bool ClashPoint(Point movePoint) {
```
Intersect

blowin commented on Snake/Extension/DirectionsExtension.cs:
```diff
@@ -0,0 +1,20 @@
+
+namespace GameSnake.Extension {
+    public static class DirectionsExtension {
+        public static Directions GetDirection(this ConsoleKey key) {
```
Хорошее решение, лучше назвать ToDirection

Stan-Kudri committed:
Rename methods:
  1)GetDirection -> ToDirection;
  2)ClashPoint -> Intersect.
